model_compression_toolkit/pytorch/back2framework/model_builder.py

```diff
@@ -0,0 +1,196 @@
+        else:
+            out_tensors_of_n = op_func(*input_tensors + op_call_args, **functional_kwargs)
+
+    # Add a fake quant node if the node has an activation threshold.
+    if mode == ModelBuilderMode.QUANTIZED and n.is_activation_quantization_enabled():
+        fake_quant = n.activation_quantization_cfg.generate_quantization_node()
+
+        if fake_quant is None:
+            raise Exception('Layer is meant to be quantized but fake_quant function is None')
+        out_tensors_of_n = fake_quant(out_tensors_of_n)
+
+    return out_tensors_of_n
+
+
+class PytorchModelBuilder(torch.nn.Module):
+    """
+    Class for reconstructing a Pytorch model from a graph
+    """
+    def __init__(self, graph: Graph,
+                 mode: ModelBuilderMode = ModelBuilderMode.QUANTIZED,
+                 append2output: List[Any] = None):
+        """
+        Construct a Pytorch model.
+        Args:
+            graph: Graph to build its corresponding Pytorch model.
+            mode: Building mode. Read ModelBuilderMode description for more info.
+            append2output: List of nodes or OutTensor objects.
+        """
+        super(PytorchModelBuilder, self).__init__()
+        self.graph = graph
+        self.mode = mode
+        self.node_sort = list(topological_sort(graph))
+        self.nodes_dict = {}
+        self.append2output = append2output
+        for n in self.node_sort:
+            if inspect.isclass(n.layer_class) and issubclass(n.layer_class, torch.nn.Module):
+                self.add_module(n.name, node_builder(n))
+                # setattr(self, n.name, node_builder(n))
+
+    def forward(self,
+                *args: Any) -> Any:
+        """
+
+        Args:
+            inputs: list of input tensors to model
+
+        Returns:
+            torch Tensor/s which is/are the output of the model logic.
+
+        """
+        node_to_output_tensors_dict = dict()
+        for n in self.node_sort:
+            input_tensors = build_input_tensors_list(n,
+                                                     self.graph,
+                                                     args,
+                                                     node_to_output_tensors_dict)
+            out_tensors_of_n = run_operation(n,  # Run node operation and fetch outputs
+                                             input_tensors,
+                                             op_func=getattr(self, n.name) if hasattr(self, n.name) else n.layer_class,
+                                             mode=self.mode)
+            if isinstance(out_tensors_of_n, list):
+                node_to_output_tensors_dict.update({n: out_tensors_of_n})
+            else:
+                node_to_output_tensors_dict.update({n: [out_tensors_of_n]})
+
+        output = []
+        if self.append2output:
+            for n in self.append2output:
+                out_tensors_of_n = node_to_output_tensors_dict.get(n)
+                if len(out_tensors_of_n) > 1:
```

Comment:
1.Can we replace this if-else to output.extend(out_tensors_of_n).
2. Lines 157-161 can be made a function.
3. Check if this is similar to Keras and let make a function in common

Reply:
1. No. Because if one of the outputs is a list, we want it to remain a list. For example: [1, 2, [3, 4]]
2. A function for these 4 lines?
3. It is not the same in Keras